jsoniter-scala-benchmark/shared/src/main/scala/com/github/plokhotnyuk/jsoniter_scala/benchmark/NinnyFormats.scala

```diff
@@ -1,10 +1,53 @@
 package com.github.plokhotnyuk.jsoniter_scala.benchmark
 
 import io.github.kag0.ninny._
-import io.github.kag0.ninny.ast.JsonObject
+import io.github.kag0.ninny.ast._
 import scala.util.Failure
+import OpenRTB._
+import com.github.plokhotnyuk.jsoniter_scala.core.JsonValueCodec
+import com.github.plokhotnyuk.jsoniter_scala.core.JsonReader
+import com.github.plokhotnyuk.jsoniter_scala.core.JsonWriter
+import GoogleMapsAPI._
+import java.time.Instant
+import java.time.OffsetDateTime
+import java.time.ZoneOffset
 
 object NinnyFormats {
+implicit val jsoniterWritingCodec = new JsonValueCodec[JsonValue] {
+    def decodeValue(in: JsonReader, default: JsonValue) = ???
+    def encodeValue(x: JsonValue, out: JsonWriter) = x match {
+        case JsonObject(values) => 
+            out.writeObjectStart()
+            values.foreach{case (k, v) => 
+                out.writeKey(k)
+                encodeValue(v, out)
+            }
+            out.writeObjectEnd()
+        case JsonDecimal(preciseValue) => 
+          if(preciseValue.precision == 0)
+          out.writeVal(preciseValue.toBigInt)
+          else
+          out.writeVal(preciseValue)
+        case JsonDouble(value) => 
+          if(value % 1 == 0)
+          out.writeVal(BigDecimal(value).toBigInt)
```

Comment:
a really inefficient way to trim a trailing `.0` lol

Reply:
I think a more efficient option is:
```
val longValue = value.toLong
if (longValue.toDouble == value) out.writeVal(longValue)
else out.writeVal(value)
```

Reply:
good point, the majority of cases are probably where `value < Long.MaxValue` as well